_studio/shared/src/libmfx_core.cpp

```diff
@@ -1508,6 +1507,12 @@ mfxStatus CoreDoSWFastCopy(mfxFrameSurface1 *pDst, mfxFrameSurface1 *pSrc, int c
             MFX_CHECK_STS(sts);
             break;
         }
+case MFX_FOURCC_A2RGB10:
+            roi.width *= 4;
+
+            sts = FastCopy::Copy(pDst->Data.B, dstPitch, pSrc->Data.B, srcPitch, roi, copyFlag);
```

Comment:
why we will receive ```nullptr``` in case of a2rgb? shouldn't all 4 pointer (```Data.A```, ```Data.R```, ```Data.G```, ```Data.B```) pointing to exactly same location?

Reply:
Saying this I'm referring to our doc, which says "The application has to specify pointers to all color channels even for packed formats. For example, for YUY2 format the application has to specify Y, U and V pointers. For RGB32 – R, G, B and A pointers." @dmitryermilov @onabiull your thoughts?

Reply:
@mgonchar this is good point, thanks!
If so, then it will be enough to add support for x2rgb10.